docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
 :::
 
-:::danger
-Taichi doesn't support building with `clang++-12` and above yet.
-`clang++-10` is highly recommended but if you're on a newer (>=11) macOS,
-building with `clang++-11` also works through `brew install llvm@11` and `export CXX=/opt/homebrew/opt/llvm@11/bin/clang++`.
-:::
+## Introduction
+
+  This installation guide covers the following:
+
+  - Prerequisites for building Taichi from source
+  - Building Taichi from source
+  - Installing optional dependencies
+  - Troubleshooting and debugging
+  - Frequently asked questions
 
 :::note
-On Linux, some additional packages might be required to build Taichi. E.g., on Ubuntu 20.04, you may need `libxi-dev` `libxcursor-dev` `libxinerama-dev` `libxrandr-dev` `libx11-dev` `libgl-dev`. please check the output of of CMake when building from source.
+
+Installation instructions vary depending on which operating system (OS) you are using. Choose the right OS or platform before you proceed.
+
 :::
 
-3. LLVM: Make sure you have version 10.0.0 installed. Taichi uses a **customized LLVM**, which we provided as binaries depending on your system environment. Note that the pre-built binaries from the LLVM official website or other sources may not work.
-   - [LLVM 10.0.0 for Linux](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_linux_patch2/taichi-llvm-10.0.0-linux.zip)
-   - [LLVM 10.0.0 for macOS (excluding M1)](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-macos.zip)
-   - [LLVM 10.0.0 for macOS (M1 chip)](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_m1/llvm-10.0.0-m1.zip)
-   - [LLVM 10.0.0 for Windows MSVC 2019](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-msvc2019.zip)
+## Prerequisites
 
+|              OS              |                                                                  Windows/macOS/Ubuntu/Arch Linux/Other Linux distributions                                                                  |
+|:----------------------------:|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|            Python            | 3.6/3.7/3.8/3.9    <blockquote>We recommend installing Python from [Miniforge](https://github.com/conda-forge/miniforge/#download) conda if you are on a MacBook with M1 chip.</blockquote> |
+|            Clang++           |                                                                                       8≥ Clang++ <12                                                                                       |
+|             LLVM             |                                                                                            10.0.0                                                                                           |
```

Comment:
Should be `Taichi's custom-built LLVM 10.0.0`. By the way, this document still doesn't tell the difference between LLVM and clang. The first time I read this, I wondered, why can't I simply `brew install llvm@10`?

Reply:
@ailzhang Please could you shed some light on this?

Reply:
> The first time I read this, I wondered, why can't I simply brew install llvm@10 
The main reason is we're not sure if llvm@10 really exist :P  https://formulae.brew.sh/formula/llvm Somehow version 10 is skipped.

Reply:
It doesn't matter how users install llvm 10. My concern is that we should highlight this is a * Taichi's custom-built* LLVM.

Reply:
I don't think this comment is resolved. If we simply list "LLVM 10.0.0" here, why can't user just install the official one from repositories like apt.llvm.org, or Arch Linux' AUR?